### ISSUE TITLE
`train_sft.py` / `train_dpo.py`: save per-step training logs into the checkpoint directory

### ISSUE THREAD

**Issue by stxxllbu** [stxxllbu/grounded-refusal#11]
## Problem

Neither training script keeps its per-step training log. After `trainer.train()`, both save only
the adapter (`trainer.save_model`) and `run_metadata.json`, a run-level summary
(`train_sft.py:147-149`, `train_dpo.py:132-134`). With `save_strategy: "no"` in both configs, no
intermediate checkpoint writes `trainer_state.json` either. The loss curve exists only as terminal
output during the run.

Week 4's loss table was copied off the terminal by hand. The Week 6 SFT retrain
(`20260913_193703_lora`) has no loss record at all (week6.md, Known follow-up 3).

## Why before Week 7

DPO's logged metrics (loss, `rewards/chosen`, `rewards/rejected`, `rewards/margins`,
`rewards/accuracies`) are the main way to tell whether a DPO run trained sensibly. If the first
DPO run doesn't save them, that evidence is gone once the terminal closes.

## Fix

After `trainer.train()` in both scripts, write the trainer's log history into the checkpoint
directory next to `run_metadata.json`:

    trainer.state.save_to_json(str(Path(output_dir) / "trainer_state.json"))

`trainer.state.log_history` holds every logged step (every `logging_steps: 2`).

Also add `!checkpoints/*/trainer_state.json` to `.gitignore`, next to the existing
`run_metadata.json` exception, so the log is committed with each run even though the weights are
not. It is a small JSON file.

## Out of scope

The existing Week 4 and Week 6 runs can't be backfilled; their logs weren't saved.
